Prepare version 2.7.1.

1uc opened the pull request:
This time we ran the deployment with the HighFive Spack recipe pointing `highfive@2.7.1` at commit `11e45f00303305dd409a5c8865a0e9dacce3cd5d`.

pramodk commented:
I know you launched the Spack/deployment CI somewhere but I don't find a link by looking at https://github.com/BlueBrain/spack/pulls.  I guess it was somewhere internal one?

matz-e reviewed (approved):
@pramodk see here: https://github.com/BlueBrain/spack/pull/1922/commits/ea1c719b1077c979bcbbf1da6032e0f3dc3372d0

pramodk commented:
Ah yes! 👌 (my eyes were just scanning the first word of the PR titles!)

1uc commented:
Second attempt for `2.7.1`. We've fixed the obvious build issues. There's still #724 to my understanding they've pinned `highfive@2.6.2`, a couple of issues will be fixed by this release, at least this release will help reduce the number of errors related to bugs we've already fixed.

The BBP Spack CI ran through when setting
```
version("2.7.1", commit="2ffb9435d8acf0701d1a30294c94c6a8d40de7a8")
```

1uc commented:
Third attempt, the Spack CI ran clean for 053ea16 which is #728.
